tests/std/tests/P0122R7_span_death/test.cpp

```diff
@@ -174,16 +178,47 @@ void test_case_algorithm_incompatible_transposed() {
 }
 
 void test_case_constructor_first_count_incompatible_extent() {
-    span<int, 3> sp(begin(globalArray), size(globalArray)); // span construction from invalid range as count != extent
+    span<int, 3> sp(begin(globalArray),
+        size(globalArray)); // Cannot construct span with static extent from range (first, count) as count != extent
     (void) sp;
 }
 
 void test_case_constructor_first_last_incompatible_extent() {
-    span<int, 3> sp(
-        begin(globalArray), end(globalArray)); // span construction from invalid range as last - first != extent
+    span<int, 3> sp(begin(globalArray), end(globalArray)); // Cannot construct span with static extent from range
+                                                           // (first, last) as last - first != extent
+    (void) sp;
+}
+
+void test_case_constructor_range_incompatible_extent() {
+    vector<int> v(begin(globalArray), end(globalArray));
+    span<int, 3> sp(v); // Cannot construct span with static extent from range r as std::ranges::size(r) != extent
     (void) sp;
 }
 
+void test_case_constructor_span_incompatible_extent() {
+    span<int, 5> sp(begin(globalArray), end(globalArray));
+    span<int, 3> sp2(sp); // Cannot construct span with static extent from other span s as s.size() != extent
+    (void) sp2;
+}
+
+void test_case_explicit_constructor_first_count() {
+    FunctionTakingSpan<int, 5>({begin(globalArray), 5}); // Constructor is explicit
+}

```

Comment:
Please verify that this test compiles and passes (you can compile it manually, even without the automated infrastructure). It appears that you've misunderstood the purpose of this test. The "death tests" in `add_death_tests` verify that `_STL_VERIFY` checks actually terminate the program at runtime (that is, their termination leads to the test passing, via special machinery). We also have a couple of "positive" tests like `test_case_operator_advance_value_initialized_iterator_zero()`, where we expect the program not to terminate. (Observe that in `main()`, these are added in different sections.)

Here, it appears that you're adding tests which should fail to compile, due to span's constructor being explicit. This will cause the test to fail to compile (which is bad).

If you want to verify the explicitness of these constructors, that should be done in `P0122R7_span` (not `P0122R7_span_death`) via SFINAE.

Reply:
Removed the function as it is not needed